File: examples/js/countwords.js
function count_words(text) {
  let count = 0
  let in_whitespace = true

  for (let char of text) {
    if (char == " " || char == "\n") {
      in_whitespace = true
    } else if (in_whitespace) {
      count += 1
      in_whitespace = false
    }
  }

  return count
}

console.assert(count_words("Hello world") == 2)
console.assert(count_words("   extra   spaces   ") == 2)
console.assert(count_words("") == 0)

File: examples/js/countwords.py
def count_words(text):
    count = 0
    in_whitespace = True

    for char in text:
        if char == " " or char == "\n":
            in_whitespace = True
        elif in_whitespace:
            count += 1
            in_whitespace = False

    return count

assert count_words("Hello world") == 2
assert count_words("   extra   spaces   ") == 2
assert count_words("") == 0